addons/mail/static/src/messaging/component/activity_mark_done_popover/activity_mark_done_popover.js

```diff
@@ -63,11 +63,16 @@ class ActivityMarkDonePopover extends Component {
      * @private
      */
     async _onClickDoneAndScheduleNext() {
+        // We need to store chatter as activity will be deleted by action.
+        const chatter = this.activity.chatter;
         const action = await this.activity.markAsDoneAndScheduleNext({
             feedback: this._feedbackTextareaRef.el.value,
         });
         this.env.do_action(action, {
-            on_close: () => this.activity.chatter.refreshActivities(),
+            on_close: async () => {
+                // If we don't wait for it, errors are not propagated.
+                await chatter.refreshActivities();
+            },
```

Comment:
I think all of this logic should move to entities, so that there are no `await` in components.
Also it makes sense for this component to not being aware of chatter, notably because it should also work in kanban view. In fact, chatter/activities relation should instead become a thread/activities relation (or better, a record/activities relation, if we introduce an entity Record... but that's for a future change, in the meantime thread/activities look good).

Still, in the scope of this PR, I'm good with keeping relation chatter/activities, and just move that in the realms of entities.

Reply:
I think it could be nice to work on the kanban widget ASAP as there is legacy code in `mail/basic_view.js` only because of him